.github/workflows/auto-merge.yml

```
@@ -0,0 +1,35 @@
+      run: |
+        pr_checks=$(gh pr view ${{ github.event.pull_request.number }} --json statusCheckRollup -q ".statusCheckRollup.status")
+        if [[ "$pr_checks" == "SUCCESS" ]]; then
+          echo "PR passed checks, ready to merge"
+        else
+          echo "PR has failed checks, not merging"
+          exit 1
+        fi
+
+    - name: Merge PR if conditions met
+      run: |
+        gh pr merge ${{ github.event.pull_request.number }} --merge --admin
```

yamllint

[error] no new line character at the end of file

(new-line-at-end-of-file)